lib/inventory/inventory_api.rb

```diff
@@ -375,4 +409,86 @@ def to_h
       hash
     end
   end
+
+  class CanonicalPath
+    attr_reader :tenant_id
+    attr_reader :feed_id
+    attr_reader :environment_id
+    attr_reader :resource_ids
+    attr_reader :metric_id
+    attr_reader :resource_type_id
+    attr_reader :metric_type_id
+
+    def initialize(hash)
+      fail 'At least tenant_id must be specified' if hash[:tenant_id].to_s.strip.length == 0
+      @tenant_id = hash[:tenant_id]
+      @feed_id = hash[:feed_id]
+      @environment_id = hash[:environment_id]
+      @resource_type_id = hash[:resource_type_id]
+      @metric_type_id = hash[:metric_type_id]
+      @resource_ids = hash[:resource_ids]
+      @metric_id = hash[:metric_id]
+    end
+
+    def self.parse(path)
+      fail 'CanonicalPath must not be nil or emtpy' if path.to_s.strip.length == 0
+      tmp = path.split('/')
+      hash = {}
+      tmp.each do |pair|
+        (key, val) = pair.split(';')
+        case key
+        when 't'
+          hash[:tenant_id] = val
+        when 'f'
+          hash[:feed_id] = val
+        when 'e'
+          hash[:environment_id] = val
+        else
+          parse_other key, val, hash
+        end
+      end
+      CanonicalPath.new(hash)
+    end
+
+    def ==(other)
+      self.eql?(other) || other.class == self.class && other.state == state
+    end
+
+    def to_s
+      ret = "/t;#{@tenant_id}"
+      ret += "/f;#{@feed_id}" unless @feed_id.nil?
+      ret += "/e;#{@environment_id}" unless @environment_id.nil?
+      ret += "/rt;#{@resource_type_id}" unless @resource_type_id.nil?
+      ret += "/mt;#{@metric_type_id}" unless @metric_type_id.nil?
+      ret += "/m;#{@metric_id}" unless @metric_id.nil?
+      ret += resources_chunk.to_s
+      ret
+    end
+
+    private
+
+    def self.parse_other(key, val, hash)
```

Comment:
On 18 Feb 2016, at 11:31, Jirka Kremser wrote:

>> +    end
>> +
>> +    def to_s
>> +      ret = "/t;#{@tenant_id}"
>> +      ret += "/f;#{@feed_id}" unless @feed_id.nil?
>> +      ret += "/e;#{@environment_id}" unless @environment_id.nil?
>> +      ret += "/rt;#{@resource_type_id}" unless 
>> @resource_type_id.nil?
>> +      ret += "/mt;#{@metric_type_id}" unless @metric_type_id.nil?
>> +      ret += "/m;#{@metric_id}" unless @metric_id.nil?
>> +      ret += resources_chunk.to_s
>> +      ret
>> +    end
>> +
>> +    private
>> +
>> +    def self.parse_other(key, val, hash)
>
> btw. this is here because of the Rubocop complaining about cyclomatic 
> complexity > 6 . So the `self.parse` handles the first 3 cases and 
> this method the rest 4.. I know, this smells. But I didn't find any 
> other way to do that. Perhaps there could be an exception for the 
> `case-when` statements, because it would me more readable to have it 
> all in 1 method.

Yes, don't play such tricks.
there is a `.rubocop` file that can be tuned.